buf/registry/module/v1beta1/commit_service.proto

```diff
@@ -193,41 +193,28 @@ message CreateCommitsRequest {
     repeated DepNode dep_nodes = 3;
   }
 
-  // The pointers to the content for the Modules that should have Commits created for them.
-  //
-  // Each ModuleNode must have a unique ModuleRef.
-  // A commit will be created for each ModuleNode.
-  repeated ModuleNode module_nodes = 1 [(buf.validate.field).repeated.min_items = 1];
-  // Blobs for the FileNodes referenced by module_nodes that are not present on the server.
-  //
-  // Only Blobs that were returned as missing from GetMissingBlobDigests need to be sent.
-  // Other Blobs already exist on the server, and will be ignored.
-  //
-  // If a FileNode within module_nodes has a Blob that is not on the server, and is not
-  // within missing_blobs, an error will be returned.
-  repeated buf.registry.storage.v1beta1.Blob missing_blobs = 2;
-  // The names of Branches that should be associated with this Commit.
-  //
-  // If a Branch currently exists on the associated Module with a name, this existing
-  // Branch will be used. Otherwise, a new Branch will be created for the corresponding name.
-  //
-  // If empty, the default branch is assumed as the only branch.
-  repeated string branch_names = 3 [(buf.validate.field).repeated.items = {
-    string: {max_len: 250}
-  }];
-  // The names of Tags that should be associated with this Commit.
-  //
-  // If a Tag currently exists on the assocated Module with a name, the RPC will error, however
-  // this will change in the future when we allow Tags to move. If the Tag does not
-  // currently exist, a new Tag will be created for each name.
-  repeated string tag_names = 4 [(buf.validate.field).repeated.items = {
-    string: {max_len: 250}
-  }];
-  // Associated VCS commit information.
-  //
-  // If there are already VCSCommits on the associated Module with a given hash, this
-  // will result in an error. Otherwise, a new VCSCommit is created.
-  repeated AssociatedVCSCommit associated_vcs_commits = 5;
+  message Value {
```

Comment:
This was primarily hoisted into a `Value` so that we can batch create `Commits` for multiple `ModuleSet`s in one RPC. Without this, I am forced to process one `ModuleSet` at a time.

Reply:
I want to understand the call pattern of this a bit better - how will I end up with `list<pair<list<module>, list<vcs_commit>>>`? Ie what is the repeated pair between `list<module> and list<vcs_commit>` for? Are we uploading multiple workspaces at once/would someone concievably want to do this?

Reply:
> Are we uploading multiple workspaces at once

Potentially, yes. 

Concretely, when we look at a Git repository, we have a Git commit with a workspace(ModuleSet) on it. Let's say I have several Git commits to upload workspaces for. In the current API, I can upload the entire workspace, with associated Git commits (i.e., squashing by identical digests), in one shot. But I can't upload multiple workspaces across Git commits where they are not identical by digest (i.e., I can't squash them) in a single shot. Batching it in the way proposed allows us to receive multiple workspaces that are not identical. 

This is not a requirement, I can always just fanout requests instead. But I figure we can't change this API later, so if we don't batch today, we are locked into that (unless we add another RPC later).

Reply:
OK makes sense.